go-controller/pkg/network-controller-manager/netattachdefcontroller.go

```diff
@@ -0,0 +1,544 @@
+package networkControllerManager
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	nettypes "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	netattachdefinformers "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/informers/externalversions"
+	networkattachmentdefinitioninformerfactory "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/informers/externalversions"
+	networkattachmentdefinitionlisters "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/client/listers/k8s.cni.cncf.io/v1"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/metrics"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/syncmap"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/util"
+	"golang.org/x/time/rate"
+)
+
+const (
+	// maxRetries is the number of times a object will be retried before it is dropped out of the queue.
+	// With the current rate-limiter in use (5ms*2^(maxRetries-1)) the following numbers represent the
+	// sequence of delays between successive queuings of an object.
+	//
+	// 5ms, 10ms, 20ms, 40ms, 80ms, 160ms, 320ms, 640ms, 1.3s, 2.6s, 5.1s, 10.2s, 20.4s, 41s, 82s
+	// maxRetries = 15
+
+	controllerName  = "net-attach-def-controller"
+	avoidResync     = 0
+	numberOfWorkers = 2
+	qps             = 15
+	maxRetries      = 10
+)
+
+type BaseNetworkController interface {
+	Start(ctx context.Context) error
+	Stop()
+	GetNetworkName() string
+}
+
+type NetworkController interface {
```

Comment:
I named this way, because in downstream, we need net-attach-def of default networks and need to handle nad events for default networks.  That is why intentionally not naming them as secondary.